### PR TITLE
Add support for searching for issues and pull requests

### PR DESCRIPTION
**Pull request by hanzei**

### CODE REVIEW

File: github.ts [hunk 1]
```diff
@@ -76,6 +76,32 @@ export async function queryNotifications({
...
+export async function querySearchIssues({
+  query,
+}: QueryProviderEvent): Promise<any[]> {
+  let api = await GithubApi.fromConfig();
+
+  let queryFilter = query.filter.find((f) => f.prop === "query");
+  if (!queryFilter) {
+    throw Error("No 'query' specified, this is mandatory");
+  }
+  query.filter = query.filter.filter((f) => f.prop !== "query");
```
Comment (hanzei) [inline github.ts:88]: Using a filter to pass an argument seems uncommon for queries. Is there another way to pass an argument?

Comment (zefhemel) [reply github.ts:88]: Yeah it's a bit strange perhaps, but now the only way to do it.

File: README.md [hunk 2]
```diff
@@ -21,10 +21,12 @@ To configure, add a `githubToken` key to your `SECRETS` page, this should be a [
 
 ## Query sources
 
-* `gh-event` required filters in the `where` clause:
+* `gh-event` List events of a user
     * `username`: the user whose events to query
-* `gh-pull`
+* `gh-pull`: List pull requests in a repository
     * `repo`: the repo to query PRs for
+* `gh-querySearchIssues`: Search for issues and pull requests
```
Comment (zefhemel) [inline README.md]: It seems this should be `gh-search-issue`?

**Review by zefhemel — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (hanzei) [reply README.md]: Good catch, fixed

**Comment by zefhemel** [comment]
Awesome!

**Pull request merged**